### PR TITLE
Add a filter to hide/disable the admin controls 

### PR DESCRIPTION
**Pull request by bernattorras**
### Description

As detailed in #43, it would be nice to be able to hide the Safety Net admin page (Tools > Safety Net) using a filter. This way, any developer (or us) could disable it easily by using the following code:

`add_filter( 'safety_net_hide_admin', '__return_true' );`

### CODE REVIEW

**Comment by NickGreen** [comment]
Most of the functions that you've allowed to be filtered out there are functions that we actually want to run, regardless of if the admin controls are showing or not. For example, `add_filter( 'wp_mail', __NAMESPACE__ . '\stop_emails', 10, 1 );` is the function that stops emails. 

I think the admin tools that he meant are the actual buttons and toggle at **Tools > Safety Net**, as seen here:
Screen Shot on 2022-11-17 at 14:29:39

So what you'll probably want to do is only include the hooks in that filterable function which render the wp-admin menu item, the link on the plugins page, and the one that renders the Safety Net Tools page.

**Comment by bernattorras** [comment]
Thanks for the head-ups @NickGreen! Yeah, I missed these filters! 

I'm a bit doubtful though about just hiding the controls... Would it make sense to have the Safety Net page available in the admin if you can't perform any actions from there? What would be the benefit? To check the possible features that aren't available for the user? 

Markup 2022-11-22 at 13 25 07

I think that since we already display the admin notice in the backend (and pause the emails, scheduled actions, etc), we could directly hide the entire Safety Net options page, but that's just my opinion. :) 

What do you think @nate-allen @NickGreen? Could you share your thoughts and possible cases to find out the best approach?

**Comment by NickGreen** [comment]
Yeah, I agree, it would be fine to hide the whole page and the links to the page (there's one in the **Tools** menu and one on the plugin page).

The main thing is that we don't want to stop _all_ of the functions that load in admin, just the ones that render the page and menu items.

File: includes/admin.php [hunk 1]
```diff
@@ -8,17 +8,31 @@
...
+	add_action( 'admin_menu', __NAMESPACE__ . '\create_options_menu' );
+	add_action( 'admin_init', __NAMESPACE__ . '\settings_init' );
+	add_action( 'wp_ajax_safety_net_anonymize_users', __NAMESPACE__ . '\handle_ajax_anonymize_users' );
+	add_action( 'wp_ajax_safety_net_scrub_options', __NAMESPACE__ . '\handle_ajax_scrub_options' );
+	add_action( 'wp_ajax_safety_net_deactivate_plugins', __NAMESPACE__ . '\handle_ajax_deactivate_plugins' );
+	add_action( 'wp_ajax_safety_net_delete_users', __NAMESPACE__ . '\handle_ajax_delete_users' );
+	add_action( 'action_scheduler_pre_init', __NAMESPACE__ . '\pause_renewal_actions' );
+	add_action( 'admin_notices', __NAMESPACE__ . '\show_warning' );
+	add_filter( 'plugin_action_links_' . SAFETY_NET_BASENAME, __NAMESPACE__ . '\add_action_links' );
+	add_filter( 'wp_mail', __NAMESPACE__ . '\stop_emails', 10, 1 );
```
Comment (nate-allen) [inline includes/admin.php:33]: We probably would still want to block emails, right?

File: includes/admin.php [hunk 2]
```diff
@@ -8,17 +8,31 @@
...
+
+	add_action( 'admin_enqueue_scripts', __NAMESPACE__ . '\enqueue_scripts' );
+	add_action( 'admin_menu', __NAMESPACE__ . '\create_options_menu' );
+	add_action( 'admin_init', __NAMESPACE__ . '\settings_init' );
+	add_action( 'wp_ajax_safety_net_anonymize_users', __NAMESPACE__ . '\handle_ajax_anonymize_users' );
+	add_action( 'wp_ajax_safety_net_scrub_options', __NAMESPACE__ . '\handle_ajax_scrub_options' );
+	add_action( 'wp_ajax_safety_net_deactivate_plugins', __NAMESPACE__ . '\handle_ajax_deactivate_plugins' );
+	add_action( 'wp_ajax_safety_net_delete_users', __NAMESPACE__ . '\handle_ajax_delete_users' );
+	add_action( 'action_scheduler_pre_init', __NAMESPACE__ . '\pause_renewal_actions' );
+	add_action( 'admin_notices', __NAMESPACE__ . '\show_warning' );
```
Comment (nate-allen) [inline includes/admin.php:32]: We should still show the warning that Safety Net is enabled

File: includes/admin.php [hunk 3]
```diff
@@ -8,17 +8,31 @@
...
+	}
+
+	add_action( 'admin_enqueue_scripts', __NAMESPACE__ . '\enqueue_scripts' );
+	add_action( 'admin_menu', __NAMESPACE__ . '\create_options_menu' );
+	add_action( 'admin_init', __NAMESPACE__ . '\settings_init' );
+	add_action( 'wp_ajax_safety_net_anonymize_users', __NAMESPACE__ . '\handle_ajax_anonymize_users' );
+	add_action( 'wp_ajax_safety_net_scrub_options', __NAMESPACE__ . '\handle_ajax_scrub_options' );
+	add_action( 'wp_ajax_safety_net_deactivate_plugins', __NAMESPACE__ . '\handle_ajax_deactivate_plugins' );
+	add_action( 'wp_ajax_safety_net_delete_users', __NAMESPACE__ . '\handle_ajax_delete_users' );
+	add_action( 'action_scheduler_pre_init', __NAMESPACE__ . '\pause_renewal_actions' );
```
Comment (nate-allen) [inline includes/admin.php:31]: I think we would still want to pause renewals

**Review by nate-allen — DISMISSED** [review]
I left some comments. It looks like to me that there are still some things being blocked that we would want to be active even if the admin controls are hidden.

**Comment by bernattorras** [comment]
Thanks @nate-allen! Yeah, these are the ones that I had in my mind, but I just wanted to clarify first if we still needed to load the options page (without the controls) or hide it completely. 👍

**Comment by nate-allen** [comment]
We should hide it completely, even removing "Safety Net" link from the Tools menu.

**Pull request merged**